ts/Core/Series/Series.ts

```diff
@@ -4374,6 +4378,21 @@ class Series {
             (series as any)[prop] = (preserve as any)[prop];
         });
 
+        // When switching back from sorted data to unsorted, reset indexes:
+        if (
+            options.data &&
+            options.dataSorting &&
+            options.dataSorting.enabled === false &&
+            oldOptions.dataSorting &&
+            oldOptions.dataSorting.enabled === true
+        ) {
+            options.data.forEach((point): void => {
+                if (isObject(point, true)) {
+                    point.x = point.index = void 0;
```

Comment:
It is kind of bad practice that the options are mutated so the we need to reset them here. Is there another way we could do this in the first place, for example storing a lookup table for data sorting?

Reply:
Thanks, I'll wait for that